### ISSUE TITLE
Meson: Cannot install on OmniOS due to invalid pam dir

### ISSUE THREAD

**Issue by rdmark** [Netatalk/netatalk#794]
```
Installing /home/dmark/netatalk/build/config/pam/netatalk to /etc/pam.conf/etc/pam.d
Traceback (most recent call last):
  File "/opt/local/lib/python3.11/site-packages/mesonbuild/mesonmain.py", line 186, in run
    return options.run_func(options)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/local/lib/python3.11/site-packages/mesonbuild/minstall.py", line 863, in run
    installer.do_install(datafilename)
  File "/opt/local/lib/python3.11/site-packages/mesonbuild/minstall.py", line 547, in do_install
    self.install_data(d, dm, destdir, fullprefix)
  File "/opt/local/lib/python3.11/site-packages/mesonbuild/minstall.py", line 621, in install_data
    if self.do_copyfile(fullfilename, outfilename, makedirs=(dm, outdir), follow_symlinks=i.follow_symlinks):
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/local/lib/python3.11/site-packages/mesonbuild/minstall.py", line 409, in do_copyfile
    dirmaker.makedirs(outdir, exist_ok=True)
  File "/opt/local/lib/python3.11/site-packages/mesonbuild/minstall.py", line 108, in makedirs
    self.makedirs_impl(path, exist_ok=exist_ok)
  File "/opt/local/lib/python3.11/site-packages/mesonbuild/minstall.py", line 309, in makedirs
    os.makedirs(*args, **kwargs)
  File "<frozen os>", line 215, in makedirs
  File "<frozen os>", line 225, in makedirs
NotADirectoryError: [Errno 20] Not a directory: '/etc/pam.conf/etc'

ERROR: Unhandled python OSError. This is probably not a Meson bug, but an issue with your build environment.
FAILED: meson-internal__install 
/opt/local/bin/meson install --no-rebuild
ninja: build stopped: subcommand failed.
```

**Comment by rdmark** [comment]
The target dir evaluates '/etc/pam.conf/etc' which is not a valid path. /etc/pam.conf exists but is a regular file.

**Comment by rdmark** [comment]
The troublesome logic is this I think

```
if host_os == 'sunos'
    pam_dir += '/etc/pam.conf'
else
    foreach path : pam_paths
        if fs.is_dir(path / 'etc/pam.d')
            pam_dir += path
        endif
    endforeach
endif

if with_pam != '' and pam_dir != '/'
    pam = declare_dependency(
        link_args: ['-L' + with_pam / 'lib', '-lpam'],
        include_directories: include_directories(with_pam / 'include'),
    )
else
    pam = cc.find_library('pam', dirs: libsearch_dirs, required: false)
endif
```

As a side note, Solaris 11 actually has an `/etc/pam.d` dir now (at least my configuration) so the basis for the `if host_os == 'sunos'` logic might no longer be valid? OmniOS, however, does *not* have it so I guess it is an old-school Solaris configuration.

Would it make more sense to check for the existence of `/etc/pam.d` instead?

**Comment by rdmark** [comment]
This might be a solution

```
diff --git a/meson.build b/meson.build
index d316cde7..d933138a 100644
--- a/meson.build
+++ b/meson.build
@@ -1397,14 +1397,14 @@ pam_paths = [
     '/usr/local',
 ]
 
-if host_os == 'sunos'
+foreach path : pam_paths
+    if fs.is_dir(path / 'etc/pam.d')
+        pam_dir += path
+    endif
+endforeach
+
+if pam_dir != ''
     pam_dir += '/etc/pam.conf'
-else
-    foreach path : pam_paths
-        if fs.is_dir(path / 'etc/pam.d')
-            pam_dir += path
-        endif
-    endforeach
 endif
 
 if with_pam != '' and pam_dir != '/'
```

**Comment by dgsga** [comment]
Fixed in PR #800
